server/src-lib/Hasura/Tracing.hs

```diff
@@ -237,22 +238,25 @@ extractEventContext e = do
 -- | Perform HTTP request which supports Trace headers
 tracedHttpRequest 
   :: MonadTrace m 
-  =>  HTTP.Request
+  => HTTP.Request
   -- ^ http request that needs to be made
   -> (HTTP.Request -> m a)
   -- ^ a function that takes the traced request and executes it
   -> m a
-tracedHttpRequest req f = trace (bsToTxt (HTTP.path req)) do
-  let reqBytes = case HTTP.requestBody req of
-        HTTP.RequestBodyBS bs -> Just (fromIntegral (BS.length bs))
-        HTTP.RequestBodyLBS bs -> Just (BL.length bs)
-        HTTP.RequestBodyBuilder len _ -> Just len
-        HTTP.RequestBodyStream len _ -> Just len
-        _ -> Nothing
-  for_ reqBytes \b ->
-    attachMetadata [("request_body_bytes", fromString (show b))]
-  ctx <- currentContext
-  let req' = req { HTTP.requestHeaders =
-                     injectHttpContext ctx <> HTTP.requestHeaders req
-                 }
-  f req'
+tracedHttpRequest req f = do
+  let method = bsToTxt (HTTP.method req)
+      uri = show @URI (HTTP.getUri req)
```

Comment:
Please don't mind if my doubt is silly but Don't we also have to include the pragma `{-#  LANGUAGE TypeApplicatdons  #-}` to use the `@` character for giving explicit type arguments?

Also what benefits do we have for using this over ` show ( (HTTP.getUri req) :: URI`

Reply:
TypeApplications is turned on by default in the Cabal file so we don't need it here. I'm using it because `show` is very polymorphic and if the argument type changes, we'll end up with a different `Show` instance being used. This has happened to me a few times before, where a refactoring broke some code using `show` or a similar function, and types were not enough to catch the mistake. This makes sure we can't accidentally change the type without changing the explicit application.

Reply:
Understood, Thanks for the explanation :)